### PR TITLE
Bug 1720333: Validate MAC when adding a NIC

### PR DESCRIPTION
**Pull request by pcbailey**
This PR adds real-time validation for the MAC address when adding a network interface from the 'Network Interfaces' tab.

@suomiy

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/components/vm-nics/create-nic-row.tsx [hunk 1]
```diff
@@ -157,12 +158,26 @@ export const CreateNicRow: React.FC<CreateNicRowProps> = ({
...
-          id="nic-mac-address"
-          onChange={setMacAddress}
-          value={macAddress}
-          disabled={creating || networkType === NetworkType.POD}
-        />
+        <FormGroup
+          className="kubevirt-vm-create-device-row__cell--no_bottom"
+          validationState={
+            macAddressError && macAddressError.type === ValidationErrorType.Error
+              ? macAddressError.type
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/vm-nics/create-nic-row.tsx]: this should be a message, right? Can you please create a function for this and use it also in the name?

File: frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { EntityMap } from '@console/shared';
-import { validateDNS1123SubdomainValue, ValidationErrorType } from '../common';
-import { addMissingSubject } from '../../grammar';
+import { isValidMAC } from 'kubevirt-web-ui-components';
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts]: can you please port the validation function here and remove the `kubevirt-web-ui-components` dependency?

File: frontend/packages/kubevirt-plugin/src/components/vm-nics/create-nic-row.tsx (hunk 1 above)
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/src/components/vm-nics/create-nic-row.tsx]: This shouldn't be a message, no. It should be an error type. The message property is used further down in a similar block. I can create functions for both, though.

File: frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts (hunk 2 above)
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts]: I can, but that would mean maintaining it in both locations. We already depend on `kubevirt-web-components` in the plugin, so I'm unclear as to what's gained by duplicating the functionality here? As long as there's an appropriate advantage to doing it then I'm happy to do so.

File: frontend/packages/kubevirt-plugin/src/components/vm-nics/create-nic-row.tsx (hunk 1 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/vm-nics/create-nic-row.tsx]: ah, you are right, sorry. I forgot this is an pf3 FormGroup. We should port these to pf4 soon.

>  I can create functions for both, though.

thanks

File: frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts (hunk 2 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts]: We are deprecating `kubevirt-web-ui-components`. It is an ongoing effort to move the whole code base here.

Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts]: Ah, ok. I misunderstood, then. I thought we were still going to maintain parts of the code base there. You're 100% correct, then. I'll get the validation function ported over, as well. =)

File: frontend/packages/kubevirt-plugin/src/components/vm-nics/create-nic-row.tsx (hunk 1 above)
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/src/components/vm-nics/create-nic-row.tsx]: Done!

File: frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts (hunk 2 above)
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/src/utils/validations/vm/nic.ts]: Done!

File: frontend/packages/kubevirt-plugin/src/utils/validations/vm/validations.ts [hunk 3]
```diff
@@ -1,16 +1,64 @@
...
+);
+
+const COLON_DELIMITER = ':';
+const DASH_DELIMITER = '-';
+const PERIOD_DELIMITER = '.';
+
+// Validates that the provided MAC address meets one of following formats supported by the golang ParseMAC function:
+// IEEE 802 MAC-48, EUI-48, EUI-64, or a 20-octet IP over InfiniBand link-layer address
+// https://golang.org/pkg/net/#ParseMAC
+export const isValidMAC = (mac) => {
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/utils/validations/vm/validations.ts]: can you add type for `mac` ?

Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/src/utils/validations/vm/validations.ts]: Done! =)

**Pull request merged**